conda/utils.py

```diff
@@ -218,21 +220,15 @@ def human_bytes(n):
 urlpath = url_path = path_to_url
 
 
-def md5_file(path):  # pragma: no cover
+@deprecated("23.9", "24.3", addendum="Use `conda.gateways.disk.read.compute_md5sum` instead.")
```

Comment:
I don't think we should suggest using a md5 specific tool if `file_digest(open(path, "b"), "md5")` also works.

Reply:
`compute_md5(path)` == `file_digest(open(path, "b"), "md5").hexdigest()`

Reply:
just reviewed all instances of these helper functions and replacing everything with `file_digest().hexdigest()` adds quite a mouthful and decreases overall readability so I'm thinking we probably want to keep `compute_digest`, `compute_md5`, and `compute_sh256` helper functions

Reply:
What rubs me the wrong way is us not deprecating enough here, I wouldn't mind if we'd keep `compute_digest` to wrap the `open` and call `hexdigest()`, but not have `compute_md5` and `compute_sha256`, since it's literally just `compute_digest(path, "md5")` and `compute_digest(path, "sha256")`, which isn't decreasing readability.

Reply:
Gotcha, I can get behind dropping the algorithm-specific implementations. 👍🏻